### PR TITLE
Corrected documentation to reflect issues with exclude

### PR DESCRIPTION
**Pull request by seddonym**
Described in this ticket: https://code.djangoproject.com/ticket/14645

### CODE REVIEW

File: docs/topics/db/queries.txt [hunk 1]
```diff
@@ -581,14 +580,34 @@ that were published in 2008. The entries selected by the second filter may or
...
+        Blog.objects.exclude(entry__headline__contains='Lennon').exclude(
+            entry__pub_date__year=2008)
+
+    To select all blogs that do not contain entries published with
+    *"Lennon"* that were published in 2008, you would need to make
+    two queries::
+
+        Blog.objects.exclude(
+            entry=Entry.objects.filter(
+                headline__contains='Lennon', pub_date__year=2008).values('id')
```
Comment (mjtamlyn) [inline docs/topics/db/queries.txt]: Can this query not be generated using `Q` objects?
`Blog.objects.filter(~Q(entry__headline__contains='Lennon') & ~Q(entry__pub_date__year=2008))`

Comment (seddonym) [inline docs/topics/db/queries.txt]: Unfortunately I don't think the `Q` object approach works.  See towards the bottom of the description on https://code.djangoproject.com/ticket/14645 - it reports that `Song.objects.filter(~Q(release_set__format='cd', release_set__released=1))` has the same issues as `Song.objects.exclude(release_set__format='cd', release_set__released=1)`.  I've just tried the equivalent of your suggestion above and it seems to have the same issue.

**Comment by seddonym** [comment]
Hi - just following up on this - is there anything else needed from me to get it merged?

**Comment by timgraham** [comment]
The new note seems to apply only to queries that span relationships; i.e. the existing docs are correct for non-relational querying. Could you update the note to make this distinction clear?

**Comment by seddonym** [comment]
Ok, I've updated the commit to make it less ambiguous.  Let me know if you need anything else.

**Comment by timgraham** [comment]
Did you test the queries in your examples? Unless I made a mistake, they don't appear to be correct.
```python
from datetime import date

from django.test import TestCase

from .models import Author, Blog, Entry

class Test(TestCase):
    def test_queries(self):
        Blog.objects.create(name='no entries')
        djangoproject = Blog.objects.create(name='djangoproject')
        google = Blog.objects.create(name='Google')
        music = Blog.objects.create(name='Music')
        Entry.objects.create(headline='Lennon', pub_date=date(2008, 1, 1), blog=music)
        Entry.objects.create(headline='Foo bar', pub_date=date(2008, 1, 1), blog=music)
        Entry.objects.create(headline='Something', pub_date=date(2008, 6, 1), blog=google)
        Entry.objects.create(headline='Lennon', pub_date=date(2009, 6, 1), blog=djangoproject)

        # blogs without entries published in 2008 and containing 'Lennon'
        q1 = Blog.objects.exclude(
            entry__headline__contains='Lennon',
            entry__pub_date__year=2008,
        ).order_by('-name')
        self.assertQuerysetEqual(q1, ['<Blog: no entries>', '<Blog: djangoproject>', '<Blog: Google>'])

        # blogs without entries published in 2008 and without entries containing 'Lennon'
        q2 = Blog.objects.exclude(
            entry__headline__contains='Lennon').exclude(
            entry__pub_date__year=2008,
        )
        self.assertQuerysetEqual(q2, ['<Blog: no entries>'])

        q3 = Blog.objects.exclude(
            entry=Entry.objects.filter(
                headline__contains='Lennon', pub_date__year=2008,
            ).values('id'),
        ).order_by('-name')
        self.assertQuerysetEqual(q3, ['<Blog: no entries>', '<Blog: djangoproject>', '<Blog: Google>'])
```

**Comment by seddonym** [comment]
Good spot, you're totally right.  The bit about it behaving the same as two exclude calls joined together wasn't true.  I've updated the documentation accordingly and tested it using the following gist: https://gist.github.com/seddonym/84407891a11389419c14#file-models-py

**Comment by timgraham** [comment]
merged in 6770b7ecd208a0746f181e54202fb829460c6490, thanks.

**Comment by seddonym** [comment]
Oh dear, I just noticed there was a typo in the final commit of this.  I've created a new pull request here: https://github.com/django/django/pull/4395

**Pull request closed without merging**